scipy/spatial/_hausdorff.pyx

```diff
@@ -0,0 +1,63 @@
+"""
+Directed Hausdorff Code
+
+.. versionadded:: 0.19.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy, Richard Gowers, and Max Linke, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+cimport numpy as np
+cimport cython
+from libc.math cimport sqrt
+
+__all__ = ['directed_hausdorff']
+
+@cython.boundscheck(False)
+def directed_hausdorff(double[:,::1] ar1, double[:,::1] ar2):
+
+    cdef double cmax, cmin, d
+    cdef int break_occurred
+    cdef int N1 = ar1.shape[0]
+    cdef int N2 = ar2.shape[0]
+    cdef int data_dims = ar1.shape[1]
+    cdef unsigned int i, j, k
+    cdef unsigned int i_store, j_store, i_ret, j_ret = 0
+    cdef long[:] resort1, resort2
+
+    # shuffling the points in each array generally increases the likelihood of
+    # an advantageous break in the inner search loop and never decreases the
+    # performance of the algorithm
+    resort1 = np.arange(N1)
+    resort2 = np.arange(N2)
+    np.random.shuffle(resort1)
+    np.random.shuffle(resort2)
```

Comment:
Ok, I'll have to be careful with this -- I had just removed seeding behaviour in response to comments from @kain88-de, so there's probably something I'm not fully understanding (random number stuff can be tricky).